local-volume/README.md

```diff
@@ -22,66 +22,43 @@ directories by creating and cleaning up PersistentVolumes for each volume.
 `MountPropagation` [feature gates must be enabled starting in 1.9](#enabling-the-alpha-feature-gates).
 
-What works:
-* Everything in 1.7.
-* New StorageClass `volumeBindingMode` parameter that fixes the previous
-  issues:
+* New StorageClass `volumeBindingMode` parameter that handles the following
+  scenarios:
     * Multiple local PVCs in a single pod.
     * PVC binding is delayed until pod scheduling and takes into account all the
       pod's scheduling requirements.
 
-What doesn't work:
-* If you prebind a PVC (by setting PVC.VolumeName) at the same time that another
-Pod is being scheduled, it's possible that the Pod's PVCs will encounter a partial
-binding failure.  Manual recovery is needed in this situation.
-    * Workarounds:
-         * Don't prebind PVCs and have Kubernetes bind volumes for the same
-           StorageClass.
-         * Prebind PV upon creation instead.
-
 ### 1.7: Alpha
 
-What works:
-* Create a PV specifying a directory with node affinity.
+* Create a PV specifying a directory or mount point with node affinity.
 * Pod using the PVC that is bound to this PV will always get scheduled to that node.
 * External static provisioner DaemonSet that discovers local directories,
   creates, cleans up and deletes PVs.
 
-What doesn't work and workarounds:
-* Multiple local PVCs in a single pod.
-    * Goal for 1.9.
-    * No known workarounds.
-* PVC binding does not consider pod scheduling requirements and may make
-  suboptimal or incorrect decisions.
-    * Goal for 1.9.
-    * Workarounds:
-        * Run your pods that require local storage first.
-        * Give your pods high priority.
-        * Run a workaround controller that unbinds PVCs for pods that are
-          stuck pending. TODO: add link
-* The provisioner will not correctly detect mounts added after it
-  has been started without mount propogation.
-  * This issue is resolved in provisioner 2.0 when the mount propagation
-  alpha feature gate is enabled, which is available in Kubernetes 1.8+.
-  * The provisioner 1.x will detect the existance of a new directory in the
-  discovery directory. Then, it will incorrectly create a local PV with
-  the root filesystem capacity. 
-  * Provisioner 1.x workaround: Before adding any new mount points, stop
-  the provisioner daemonset, add the new mount points, start the daemonset.
-
 ### Future features
-* Local block devices as a volume source, with partitioning and fs formatting
+
 * Pod accessing local raw block device
+* Local block devices as a volume source, with partitioning and fs formatting
+* Dynamic provisioning for shared local persistent storage
 * Local PV health monitoring, taints and tolerations
 * Inline PV (use dedicated local disk as ephemeral storage)
-* Dynamic provisioning for shared local persistent storage
+
+## Provisioner Compatibility
+
+| Provisioner Version | Min K8s Version |
+| ------------------- | --------------- |
+| 2.0.0               | 1.8             |
```

Comment:
Why can't 2.0 provisioner be used with 1.7?

Reply:
Oh it's if you want to fix the mount propagation issue.  I guess it's strictly not required if you don't care about it.  Hmmm this is becoming more difficult to convey.

I imagine we'll want to say things like:
- If you want to fix the mount propgation issue, then you must use K8s 1.8 + provisioner 2.0.0
- If you want block volume support, then you must use K8s 1.10 + provisioner 2.0.1
- If you want to use beta local volume API, then you must use K8s 1.10 + provisioner 2.1.0
- When alpha local volume API is deprecated and removed, then you can't use provisioner < X.

Some dependencies are soft (ie only if you need that feature), and some may be hard (ie alpha API will be deprecated and removed, so lower versions will definitely not work anymore, and higher versions won't support alpha API).  

I am open to suggestions on how to convey this nicely.

Reply:
I agree that it is hard to describe all the various potential configurations and combinations.

I'm inclined to recommend that people use the latest version of kubernetes and the latest version of the provisioner with all the listed feature-gates, and that we describe the latest version's functions and capabilities here. 

We can point to CHANGELOG for bugfixes and addition of new features (e.g. block). We can point to KNOWN_ISSUES to explain earlier version issues, and mention which provisioner-version/k8s-version/feature-gate combination addresses a particular issue earlier.

Reply:
K I'm trying this out:
Provisioner features are in provisioner CHANGELOG.  Known issues are in KNOWN_ISSUES.
K8s features are in main README. Known issues are in KNOWN_ISSUES.